src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -376,7 +376,8 @@ TokenizeSnippets::TokenizeSnippets() {
                             size_t current_input_index = body_parameters.size();
                             for (size_t p_ind = 0; p_ind <  body_parameters.size(); p_ind++) {
                                 const auto & p = body_parameters[p_ind];
-                                if (p->get_friendly_name() == found->get_node_shared_ptr()->get_friendly_name()) {
+                                if (p->get_friendly_name() == found->get_node_shared_ptr()->get_friendly_name() &&
+                                    external_inputs[p_ind] == *found) {  // or replace only when common parent subgraph have one output
```

Comment:
It'd be great if you add comment that we check for the same output ports of parent.
I mean that we can unite body parameters only if they're connected to the same parent of Subgraph (it's already been) and to the same output port of this parent (it's added in the PR). What do you think? It's just to remember what happens here.

Thanks!

Reply:
Agree, a comment is definitely needed here.

Reply:
comments is added! thanks.